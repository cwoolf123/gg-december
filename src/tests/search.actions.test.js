
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';
import fetch from 'isomorphic-fetch';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

// We would import these from actions in real case testing.
const success = data => ({
  type: 'SEARCH',
  data,
});

// Async action creator to fetch to-dos from an API
const fetchTodos = () => dispatch => fetch('https://localhost:8000/api/books')
  .then(response =>
    // console.log('This is the response body', response.json());
    // dispatch(success(response.json()));
    response.json(),
  )
  .then((json) => {
    dispatch(success(json));
  });

describe('async actions', () => {


  let store;
  const fetchSearchData = {
          books: [
              { id: 2086,
                book_author: [
                  "Author"
              ],
                book_title: 'Example',
                book_publication_year: 1529,
                book_publication_country: "Country",
                book_publication_city: "City",
                book_pages: 104
              },
            ],
        count: 1,
        };

  beforeEach(() => {
    store = mockStore({});
  });
  afterEach(() => {
    // clear all HTTP mocks after each test
    nock.cleanAll();
  });

  it('creates SEARCH response when mock search has been completed', () => {
    // Simulate a successful response
    nock('https://localhost:8000')
      .get('/api/books') // Route to catch and mock
      .reply(200, fetchSearchData); // Mock reponse code and data

    // Dispatch action to fetch to-dos
    return store.dispatch(fetchTodos())
      .then(() => { // return of async actions
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});