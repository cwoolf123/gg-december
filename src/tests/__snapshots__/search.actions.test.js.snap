// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async actions creates SEARCH response when mock search has been completed 1`] = `
Array [
  Object {
    "data": Object {
      "books": Array [
        Object {
          "book_author": Array [
            "Author",
          ],
          "book_pages": 104,
          "book_publication_city": "City",
          "book_publication_country": "Country",
          "book_publication_year": 1529,
          "book_title": "Example",
          "id": 2086,
        },
      ],
      "count": 1,
    },
    "type": "SEARCH",
  },
]
`;
